### DISCUSSION TITLE
Thoughts wanted on issues #480/481

### DISCUSSION THREAD

**Discussion by SpenceKonde** [SpenceKonde/megaTinyCore#482] [Future Development Discussion]
I just want to call additional attention to these two issues I created and get some thoughts on how this should  be handled. We have time (I won't get to circle back to this for a while after 2.4.0) but this is a very impactful design decision relating to the Peripheral of Peripherals, the event system itself, which (alongside the CCL, with which it is intimately linked) is what makes these parts more than merely an incremental improvement over classic AVRs.

**Comment by grandaspanna** [comment]
Not familiar with github best practice - do we comment on the issues or here in the discussion?

**Reply by SpenceKonde** [reply]
IMO: If it is relevant to the solving the problem - proposals or debate on what the correct solution should look like - it should go over in issues. 

Clarification regarding the details of the the subject at hand - that's probably not going to be generally useful, lets keep it here (I'll edit in important clarifications as they reveal themselves) 

Philosophical debate about whether an event system library should exist at all should stay over here in discussions  - or on the walls of your local public restroom. Unless I've missed something fundamental, and the current implementation is far more useful than it seems - we need major usability enhancements here; EVSYS is too important of a peripheral to neglect.

**Comment by hmeijdam** [comment]
Philosophical....Is the objective to enable users to understand the benefits of the event system (flatten the learning curve), or do you assume they already know and  the library just needs to make using the event-system less "specific-part-dependent"? . Without some understanding of the different peripherals of the Tiny, I think a library is not going to make it easier. 

I compare the event system with the wires on my breadboard, connecting different electronic components together.
So A more graphical presentation of the parts peripherals with lines going from one peripheral to the other would work better for me to understand the event system.

Sort of how I also learned what the different timers can do.  if I see it visually presented I can understand and remember it a lot easier.
example :  https://github.com/dbuezas/arduino-web-timers
(don't forget to click on the link above the image to actually see the web-timer in action)
